### ISSUE TITLE
Not able to load template in python, windows.

### ISSUE THREAD

**Issue by sidhujasminder** [pallets/jinja#711]
When trying to load_template from an HTML file, the following line does the check on path.
https://github.com/pallets/jinja/blob/master/jinja2/loaders.py#L27
But this code is throwing TemplateNotFound: 'C:\path\to\template' in windows.
I wrote a testcase case which would a Jinja template from a file.

The following python code is throwing error. 
```
templateLoader = jinja2.FileSystemLoader(searchpath="/")
templateEnv = jinja2.Environment(loader=templateLoader)
TEMPLATE_FILE = os.path.join(os.path.dirname(__file__), 'content.html')
template = templateEnv.get_template(TEMPLATE_FILE)
```
For the last line of the code, I get the error. The same code is working fine on ubuntu 
---

## Expected Behavior
It should load the template from the file.
## Actual Behavior
Throwing error.

## Template Code
```jinja
Paste the template code (ideally a minimal example) that causes the issue

```

## Full Traceback
```pytb

  File "c:\python27\lib\site-packages\jnpr\jsnapy\notify.py", line 38, in notify
    template = templateEnv.get_template(TEMPLATE_FILE)
  File "c:\python27\lib\site-packages\jinja2\environment.py", line 812, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "c:\python27\lib\site-packages\jinja2\environment.py", line 774, in _load_template
    cache_key = self.loader.get_source(self, name)[1]
  File "c:\python27\lib\site-packages\jinja2\loaders.py", line 168, in get_source
    pieces = split_template_path(template)
  File "c:\python27\lib\site-packages\jinja2\loaders.py", line 31, in split_template_path
    raise TemplateNotFound(template)
TemplateNotFound: c:\python27\lib\site-packages\jnpr\jsnapy\content.html
```

## Your Environment
* Python version: 2.7.13
* Jinja version:2.8.1

**Comment by ThiefMaster** [comment]
Template paths are not necessarily filesystem paths and thus always use forward slashes. If you need to join these paths use string concatenation or `posixpath.join()`

**Comment by sidhujasminder** [comment]
@ThiefMaster Thankyou for your information. But if I use posixpath.join(os.path.dirname(__file__), 'content.html') it shows same error with **c:\python27\lib\site-packages\jnpr\jsnapy/content.html**. How the path os.path.dirname(__file__) will have slashes changed to forward slash. Any idea?

**Comment by ThiefMaster** [comment]
you should not used that at all in the template path. use it on the base 
directory of the FileSystemLoader instead and then specify a template path 
relative to that.

**Comment by sidhujasminder** [comment]
Thanks @ThiefMaster.

**Comment by ben-whale** [comment]
I have encountered this problem also. As I understand it I should be passing the base path into the FileSystemLoader and then the name of the template file into the get_template() function? This isn't made clear documentation?

As a side point why is a distinction between template paths and system paths made? Is there some documentation I could read about this?

**Comment by icfly2** [comment]
I've had this problem too, but found that it has nothing to do with the backslashes `\`.

Instead it is just very intuitive where it looks for templates. I used `env.list_templates()` for my environment `env` to figure out where it was looking and them moved from there. 

in the `get_template` part you do need forward slashes, but the output of `list_templates` makes that pretty clear.

**Comment by AdamJel** [comment]
Hi, 
in case this would be of any help to someone someday.

Windows 10 machine.

I spend last two hours solving, why I get `TemplateNotFound` error. I have read everything. I have tried everything. Nothing works, well except the error - that worked each time.

Solution: I was not in elevated cli. Yes, that's it. Once I elevated, it worked. So I had to rerun my IDE as an admin as well. It is pretty clear now, but it didn't occur to me. I was trying to figure the problem in program, rather then OS.

Perhaps, there could be some other test-error pair for forbidden OS-file-system access?
